syntax = "proto3";

import "echo_attributes.proto";

package test_messages;

message TestString {
  string value = 1 [(string_size) = 20];
}

message TestRepeatedString {
  repeated string value = 1 [(string_size) = 20, (array_size) = 10];
}

message TestUint32 {
  uint32 value = 1;
}

message TestMessageWithMessageField {
  TestUint32 message = 1;
}

message TestNestedMessage
{
	message NestedMessage
	{
		uint32 value = 1;
	}

	NestedMessage message = 1;
}

message TestNestedRepeatedMessage
{
	message NestedMessage
	{
		uint32 value = 1;
	}

	repeated NestedMessage message = 1 [(array_size) = 10];
}

message TestEmptyMessage
{
}

service TestService1
{
  option (service_id) = 1;

  rpc Method(TestUint32) returns (Nothing) { option (method_id) = 1; }
}

service TestService2
{
  option (service_id) = 2;

  rpc Search (TestString) returns (Nothing) { option (method_id) = 1; }
}
